Arduino.mk

```diff
@@ -899,6 +902,9 @@ ifndef OVERRIDE_EXECUTABLES
     AR      = $(AVR_TOOLS_PATH)/$(AR_NAME)
     SIZE    = $(AVR_TOOLS_PATH)/$(SIZE_NAME)
     NM      = $(AVR_TOOLS_PATH)/$(NM_NAME)
+
+    # Check if the default exist, otherwise try to find in PATH
+    $(foreach C,CC CXX AS OBJCOPY OBJDUMP AR SIZE NM,$(if $(wildcard $C),,$(eval $C = $$(shell type -p $$($(C)_NAME)))))
```

Comment:
does this actually work (when using which) it doesn't seem to populate the variables when i try, so you end up with: 

```
/bin/sh: 1: CC_NAME: not found
/bin/sh: 1: CXX_NAME: not found
/bin/sh: 1: AS_NAME: not found
/bin/sh: 1: OBJCOPY_NAME: not found
/bin/sh: 1: OBJDUMP_NAME: not found
/bin/sh: 1: AR_NAME: not found
/bin/sh: 1: SIZE_NAME: not found
/bin/sh: 1: NM_NAME: not found
```